syntax = "proto3";
import "google/protobuf/empty.proto";
option go_package = ".;proto";

service UserFavorite {
    rpc GetUserFavoriteList(UserFavoriteRequest) returns (UserFavoriteListResponse);
    rpc GetUserFavoriteDetail(UserFavoriteRequest) returns (google.protobuf.Empty);
    rpc CreateUserFavorite(UserFavoriteRequest) returns (google.protobuf.Empty);
    rpc DeleteUserFavorite(UserFavoriteRequest) returns (google.protobuf.Empty);
}

message UserFavoriteRequest {
    int32 userId = 1;
    int32 goodsId = 2;
}

message UserFavoriteResponse {
    int32 userId = 1;
    int32 goodsId = 2;
}

message UserFavoriteListResponse {
    int32 total = 1;
    repeated UserFavoriteResponse data = 2;
}
